mlflow/utils/git_utils.py

```diff
@@ -0,0 +1,18 @@
+from typing import Optional
+
+
+def get_git_repo_url(path: str) -> Optional[str]:
+    """
+    Obtains the url of the git repository associated with the specified path,
+    returning ``None`` if the path does not correspond to a git repository.
+    """
+    from git import Repo
+
+    try:
+        repo = Repo(path, search_parent_directories=True)
+        remote_urls = [remote.url for remote in repo.remotes]
+        if len(remote_urls) == 0:
+            return None
+    except Exception:
+        return None
+    return remote_urls[0]
```

Comment:
```suggestion
    try:
        repo = Repo(path, search_parent_directories=True)
        return next(remote.url for remote in repo.remotes, None)
    except Exception:
        return None
```

Can we use `next(..., None)` to simplify the code?